app/models/reception/resource_factory.rb

```diff
@@ -0,0 +1,122 @@
+# frozen_string_literal: true
```

Comment:
It might be worth adding some more documentation here about why you are introducing caches. I think you mentioned it the other day so I was aware but it might be slightly opaque.